crutest/src/main.rs

```diff
@@ -2185,11 +2351,149 @@ async fn perf_workload(
+
+    let stop = Arc::new(AtomicBool::new(false));
+    let byte_count = Arc::new(AtomicUsize::new(0));
+
+    let block_size = ri.block_size as usize;
+    let total_blocks = ri.total_blocks;
+    for _ in 0..cfg.io_depth {
+        let stop = stop.clone();
+        let guest = guest.clone();
+        let byte_count = byte_count.clone();
+        tokio::spawn(async move {
+            match cfg.mode {
+                RandReadWriteMode::Write => {
+                    let mut buf = BytesMut::new();
+                    buf.resize(cfg.blocks_per_io * block_size, 0u8);
+                    let mut rng = rand_chacha::ChaCha8Rng::from_entropy();
+                    rng.fill_bytes(&mut buf);
+                    let block_shift = block_size.trailing_zeros();
+                    while !stop.load(Ordering::Acquire) {
+                        let offset =
+                            rng.gen_range(0..=total_blocks - cfg.blocks_per_io);
+                        guest
+                            .write(
+                                Block::new(offset as u64, block_shift),
+                                buf.clone(),
+                            )
+                            .await
+                            .unwrap();
+                        byte_count.fetch_add(buf.len(), Ordering::Relaxed);
+                    }
+                }
+                RandReadWriteMode::Read => {
+                    let mut buf = Buffer::new(cfg.blocks_per_io, block_size);
+                    let mut rng = rand_chacha::ChaCha8Rng::from_entropy();
+                    let block_shift = block_size.trailing_zeros();
+                    while !stop.load(Ordering::Acquire) {
+                        let offset =
+                            rng.gen_range(0..=total_blocks - cfg.blocks_per_io);
+                        guest
+                            .read(
+                                Block::new(offset as u64, block_shift),
+                                &mut buf,
+                            )
+                            .await
+                            .unwrap();
+                        byte_count.fetch_add(buf.len(), Ordering::Relaxed);
+                    }
+                }
+            }
+            Ok::<(), CrucibleError>(())
+        });
+    }
+
+    let mut prev = 0;
+    for i in 0..cfg.time_secs {
+        std::thread::sleep(std::time::Duration::from_secs(1));
+        let bytes = byte_count.load(Ordering::Acquire);
+        let delta = bytes - prev;
+        prev = bytes;
+        print!(
+            "{} {}/sec ({} IOPS)  [{} secs remaining]    ",
+            if i == 0 { '\n' } else { '\r' },
+            human_bytes(delta as f64),
+            delta / (cfg.blocks_per_io * block_size),
+            cfg.time_secs - i,
+        );
+        std::io::stdout().lock().flush().unwrap();
+    }
+    stop.store(true, Ordering::Release);
+
+    let bytes = byte_count.load(Ordering::Acquire);
+    println!(
```

Comment:
This final print can sometimes get lost if there is still IO outstanding when the
timer elapses.  Might be a little nicer if the test paused for the queues to
empty before printing out a summary?  I ran this with an `io-depth` of 16 and I
got a bunch of `WARN exiting client task is ignoring message` message which
buried my results.

Reply:
Oh, I think we might want to require this.

If I have a bunch of IO outstanding, it's possible to exit the test early
with IO inflight such that some downstairs did get the IO and some did
not.  Also, if any extents are dirty, we don't actually know what they
have. All this means the next time you try to use that region set, it will have to
reconcile the three downstairs.

I did it with a region like this:
```
./target/release/dsc start --create --cleanup --extent-size=16 --extent-count=1024 --block-size=4096
```

Then, running a write workload to it.